### ISSUE TITLE
Slow. Very slow.

### ISSUE THREAD

**Issue by sirrooster** [Feodor2/Mypal68#511]
After a while of usage the browser slows down extremely

**Comment by NS-Clone** [comment]
after one of mypal process reach 1GB size
it usually highly slow downs

**Comment by Feodor2** [comment]
It is but not so much for me, I think this is a common firefox drawback, I don't know where to look and what to do with this
For begins how do you measure the slow?

**Comment by krzysiobal** [comment]
Then maybe tell us, why it requires 1GB of memory and 50% of cpu usage if there is no single tab opened?
![mypal](https://github.com/user-attachments/assets/7fc1560e-9ff4-48c2-ab8a-a729ac8537d8)

**Comment by bluedragon-cairo** [comment]
might be a memory leak inherited from Firefox

**Comment by Feodor2** [comment]
Your pic is a rubbish you know, may be first to tell what you did, which sites opened and else, all steps prior to show the empty window, and how much windows actually

**Comment by sirrooster** [comment]
it happens on many sites.
e.g. Facebook is getting very slow, pictures don't load and I get a prompt to disable a script. After disableing it's getting faster again, but many functions on FB don't work anymore.

**Comment by Feodor2** [comment]
https://github.com/Feodor2/Mypal68/issues/439#issuecomment-2103326757
What cpu do you have. Give an actual link I can check it today on the pentium dialcore 6500.

**Comment by zanud** [comment]
I can add something (perhaps not new) to complicate the picture.

When I am browsing Aliexpress, Mypal is very, very, very slow. But only if I am using my usual profile. When I am using a completely new profile, Aliexpress is reasonably fast.

At the same time, I can't imagine which settings, that I changed in the usual profile, may produce such drastic effect - almost everything was changed via "Options" dialog.

**Comment by Ariel-Schnee** [comment]
Let's just face it. Mypal 68.14.(anything) keeps crashing. That's why I went back to Mypal 68.13.9b (32-bit) instead. It doesn't crash. It keeps working.



---

Signature - SFW - https://www.fanficparadise.com/arielschnee/sig3.htm
Signature - NSFW - https://www.fanficparadise.com/arielschnee/sig3_NSFW.htm


  ----- Original Message -----

**Comment by FlorianisonGitHub** [comment]
Slow on 2003 Celeron processor with Dell Dimension 2400. Any ways to optimize it

**Comment by sirrooster** [comment]
> [#439 (Kommentar) ](https://github.com/Feodor2/Mypal68/issues/439#issuecomment-2103326757) Welche CPU hast du? Geben Sie einen aktuellen Link an. Ich kann ihn heute auf dem Pentium Dialcore 6500 überprüfen.

![cpu](https://github.com/user-attachments/assets/16a6634f-337b-4d02-8ca4-ac00fff709f5)

What Link are you asking for??

**Comment by zanud** [comment]
> What Link are you asking for??

He asked for one or a few links where:
> it happens on many sites.

Site must be accessible without registration, so your link to Facvebook is not suitable.

**Comment by sirrooster** [comment]
Facebook is the only site where the advice of the script error appears.
But it's getting slow on many sites: Ebay, Twitter, my government tax-pay site.
All these sites are with registration.
I will let you know if I find a slow-down site without registration.

**Comment by zanud** [comment]
> I will let you know if I find a slow-down site without registration.

Youtube :smile:
Each page opens quite fast. But when I try scroll down to user comments, it eats essential amount of time.
And the worst happens when I am trying to expand any conversation under user comment - browser freezes approximately for a minute. Why??? During this time it only waits for a server response in the asynchronous call. So why this call freezes a whole browser?

**Comment by Feodor2** [comment]
I do not confirm this

**Comment by Forlix** [comment]
This one can be closed. I've had periodic freezes every ten seconds or so and for several seconds at a time which was very annoying. They seemed to build up from browsing activity. Happy to report this has been resolved with 5b. Overall performance and start-up time have improved dramatically and my page file is no longer being maltreated. Thanks.

**Comment by krzysiobal** [comment]
CPU usage jumps quickly between 0% and 50% on this site, despite nothing being animated or done in the background:
https://parcelsapp.com/en/tracking/
Any good reason for this?

**Comment by NS-Clone** [comment]
> despite nothing being animated or done in the background:

it's modern sites
it's extremely waste of CPU power
but do nothing

**Comment by NS-Clone** [comment]
also i think mypal works slihly faster with
```
/* nearest-neighbor by default */
img {
  image-rendering: crisp-edges !important;
}
```
in userContent.css

but you'll get aliasing evetywhere

**Comment by zanud** [comment]
> it's extremely wasting of CPU power
> but do nothing

Not exactly. One of ParcelsApp's scripts is running endless recursion. So it not only is wasting the CPU, but also consuming more, and more, and more memory.

Unfortunately, that script is from CloudFront, and is mangled, so it is impossible to analyze it.

**Comment by bushy555** [comment]
Appreciation post to @Feodor2 

...finally got around to installing v14.7 from v13.7.
v14.7 is running twice as fast as what 13.7 was!   Love it!!!

**Comment by NS-Clone** [comment]
ну вот браузер обновил
и субъективно все залетало... (несмотря на то что браузер вылез на полтора гига за размер физической рамы (тк запущен еще один браузер))
и вощем как всегда субъективно на слещий день все опять будет тормозить

а можот в этом виновато какоето кешированиё?
браузер не чистит какуюто еще кешу по поводу обновлений случаем?
(кеша для сайтеков умя вроде отключен)

**Comment by zanud** [comment]
> можот там можно периодически сносить какойнить Profile\startupCache ??

Можно. Более того, я периодически удаляю вообще всю ту часть профиля, которая лежит в `Local Settings\Application Data`.

Единственное что - при этом удаляются превьюшки сайтов, которые Mypal показывает на стартовой странице (`about:newtab`). Но я этой страницей не пользуюсь, у меня везде `about:blank`.

> ну и почистить можно только снеся папку с кукисами :)
Куки находятся в файле `cookies.sqlite`. А в каталоге `storage` лежит то, что Mypal называет "Данные сайтов"

**Comment by NS-Clone** [comment]
> Можно. Более того, я периодически удаляю вообще всю ту часть профиля, которая лежит в `Local Settings\Application Data`

ну умя там никто и не лежит
ибо негоже там кому либо без проса лежать :)

**Comment by zanud** [comment]
Там ничего нет, если браузер в портативном режиме используется. В этом случае у него в одном каталоге вперемешку навалено то, что у обычного разделено между `Application Data` и `Local Settings\Application Data`.

А так-то в `Local Settings\Application Data` лежит всё, что относится к кешам. В том числе и закешированные потроха самого браузера.
